### PR TITLE
Compare the full 64 bits address on Solana

### PR DESCRIPTION
**Pull request by seanyoung**
vector_new() takes an initializer arguments, which has as special value
for "no initializer". We should check the full 64 bits, rather than
casting the lower 32 bits to int.

This change exposes a bug in the old BPF target in Solana's LLVM fork. Switching 
to the newer SBF target fixes the issue.

### CODE REVIEW

**Comment by LucasSte** [comment]
> This change exposes a bug in the old BPF target in Solana's LLVM fork. Switching to the newer SBF target fixes the issue.

The LLVM compilation instructions in our docs must be updated as well.

**Comment by seanyoung** [comment]
> The LLVM compilation instructions in our docs must be updated as well.

I don't think there is anything that needs updating there, as the LLVM target are not listed. I have replaced occurrences of BPF with SBF.

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -41,14 +41,14 @@ jobs:
...
-    container: ubuntu:20.04
+    container: ubuntu:22.04
     steps:
       - name: Checkout sources
         uses: actions/checkout@v3
       - name: Install Python and git
         run: |
           apt-get update
-          apt-get install -y python3-pip git
+          apt-get install -y python3-pip git pkg-config libcairo-dev
```
Comment (LucasSte) [inline .github/workflows/test.yml:51]: These changes are also in #1298. I think it is best the leave them only in one place.

**Review by LucasSte — APPROVED** [review]

File: .github/workflows/test.yml (hunk 1 above)
Comment (seanyoung) [reply .github/workflows/test.yml:51]: That was done to make the tests pass. Once one PR is merged, the rebase will remove the duplicate code

Comment (LucasSte) [reply .github/workflows/test.yml:51]: > That was done to make the tests pass. Once one PR is merged, the rebase will remove the duplicate code

This is why these changes better suit in a separate PR.

**Review by xermicus — APPROVED** [review]

File: docs/language/expressions.rst [hunk 2]
```diff
@@ -143,7 +143,7 @@ an bitwise XOR of all function selectors in the interface. This makes it possibl
 an interface at runtime, which can be used to write a `supportsInterface()` function as described
 in the EIP.
 
-The contract code for a contract, i.e. the binary WebAssembly or BPF, can be retrieved using the
+The contract code for a contract, i.e. the binary WebAssembly or SBF, can be retrieved using the
```
[the remark this answers is not in the document]
Comment (seanyoung) [reply docs/language/expressions.rst]: I've replaced them all with `Solana SBF`. Thanks for the suggestion, this is much clearer

**Pull request merged**